### ISSUE TITLE
CMake build_shared property

### ISSUE THREAD

**Issue by uilianries** [conan-io/conan#1132]
Hi!

Conan has a good cmake support, but when I want create _shared_ option for some new package, I need parse the option. However, CMake has default option for create shared/static library.  For example:  

```python
    # conanfile.py
    def build(self):
        cmake = CMake(self.settings)
        shared = "-DBUILD_SHARED_LIBS=ON" if self.options.shared else "-DBUILD_SHARED_LIBS=OFF"
        self.run('cmake %s %s %s' % (self.conanfile_directory, cmake.command_line, shared))
        self.run("cmake --build . %s" % cmake.build_config)
```

My proposal is parse this option, in CMake class:  
```python
    # conanfile.py
    def build(self):
        cmake = CMake(self.settings, self.options)
        self.run('cmake %s %s' % (cmake.command_line, shared))
        self.run("cmake --build . %s" % cmake.build_config)
```
```python
    # cmake.py
    @property
    def build_shared(self):
        try:
            build_shared = self._options.shared
            return "-DBUILD_SHARED_LIBS=ON"
        except ConanException:
            return ""
```

Some projects are using Makefile, so will not affected by this feature.  
Most of the projects are using _shared_ as option to build shared/static library. It's a good point to create a standard.

**Comment by memsharded** [comment]
I think is doable, in fact, it could be also automated inside the ``CMake.configure()`` method.
Also, probably pass the complete conanfile object to CMake, like ``cmake = CMake(self)``, for simplicity.

**Comment by uilianries** [comment]
**update:**

Using a better approach, by current CMake version (conan 0.21):

```python
    def build(self):
        shared = {"BUILD_SHARED_LIBS": "ON" if self.options.shared else "OFF"}
        cmake = CMake(self.settings)
        cmake.configure(self, defs=shared)
        cmake.build(self)
```

So, your plan is:
```python
    def build(self):
        cmake = CMake(self)
        cmake.configure()
        cmake.build()
```
Is it correct?

**Comment by lasote** [comment]
Yes, it is.

**Comment by lasote** [comment]
It has been released in conan 0.22.0. Any feedback is welcome. Thanks.
